dashboard/app/controllers/course_offerings_controller.rb

```diff
@@ -35,6 +35,12 @@ def quick_assign_course_offerings
     render :ok, json: offerings.to_json
   end
 
+  def self_paced_pl_course_offerings
+    return head :bad_request unless current_user
+    offerings = CourseOffering.assignable_course_offerings(current_user).select {|co| co.get_participant_audience == 'teacher' && co.instruction_type == 'self_paced'}
+    render :ok, json: offerings.map(&:summarize_self_paced_pl).to_json
```

Comment:
Nit: Not sure if this is even a case that could happen but wondering if there should be safe chaining after `offerings` in the case that the query returns no results

Reply:
Yesss good call. Added!